### PR TITLE
Ini file fixes

### PR DESCRIPTION
**Pull request by akallabeth**

### CODE REVIEW

File: winpr/include/winpr/ini.h [hunk 1]
```diff
@@ -30,27 +30,123 @@ extern "C"
...
+	 *
+	 *  @return > 0 for success, < 0 for failure
+	 */
 	WINPR_API int IniFile_ReadFile(wIniFile* ini, const char* filename);
 
+	/** @brief write an ini instance to a buffer
+	 *
+	 *  @param ini The instance to use, must not be \b NULL
+	 *
+	 *  @return A newly allocates string, use \b free after use. \b NULL in case of failure
```
Comment (mfleisz) [inline winpr/include/winpr/ini.h:55]: allocate~~s~~d string

File: winpr/libwinpr/utils/ini.c [hunk 2]
```diff
@@ -304,35 +322,45 @@ static wIniFileSection* IniFile_GetSection(wIniFile* ini, const char* name)
 	return section;
 }
 
+static BOOL IniFile_SectionResize(wIniFile* ini, size_t count)
+{
+	WINPR_ASSERT(ini);
+
+	if (ini->nSections + count >= ini->cSections)
+	{
+		const size_t new_size = ini->cSections + count + 1024;
```
Comment (mfleisz) [inline winpr/libwinpr/utils/ini.c:331]: Why is the `+ 1024` (same in `IniFile_SectionKeysResize`)?
It seems we are always carrying 1024 additional empty slots around with us

**Review by mfleisz — APPROVED** [review]
2 minor comments - otherwise lgtm

File: winpr/libwinpr/utils/ini.c (hunk 2 above)
Comment (akallabeth) [reply winpr/libwinpr/utils/ini.c:331]: block allocation, always extend by `1024` if we reach the current limit

**Pull request merged**